### ISSUE TITLE
Firefox: Bad heatmap label formatting

### ISSUE THREAD

**Issue by schmich** [schmich/marinara#114]
Font is too large and is clipped. Font is still correct on the daily/weekly distribution charts.

![screen shot 2017-10-24 at 3 45 29 am](https://user-images.githubusercontent.com/1562839/31933138-deb3bb84-b86d-11e7-8440-67e3767537e1.png)

**Comment by jamielockie** [comment]
Hey, 

I've been using Marinara for a while now and love it.  I'm also new to Open Source and would like to take a shot at fixing this.

**Comment by schmich** [comment]
Sounds great! I welcome contributions. I'll try to point you in the right direction.

This issue is part of the [Firefox milestone](https://github.com/schmich/marinara/milestone/8) to get Marinara running on Firefox. The setup is still a little wonky since I haven't had many contributors:

- Install [Firefox Quantum Beta](https://www.mozilla.org/en-US/firefox/quantum/)
- Install [web-ext](https://developer.mozilla.org/en-US/Add-ons/WebExtensions/Getting_started_with_web-ext), Mozilla's tool for working with web extensions
- Fork & clone this repo, switch to the `firefox` branch where all Firefox-specific work happens
- Run `make dev-firefox` in the root directory of the repo, this will create `src/manifest.json`, a Firefox-specific extension manifest
- `cd src` and run `web-ext run` to start Firefox Quantum with the extension loaded
- Right-click the tomato in the toolbar and select "Pomodoro History" as a roundabout way of getting to the Settings page (because of #111)
- Lower the timer to `0.1`, then run and complete a Pomodoro
- Open the History page again and check the heatmap chart, you should see the screenshot above

Heatmap rendering is done with [D3](https://d3js.org/), and the code for it is in [options/charts.js](https://github.com/schmich/marinara/blob/master/src/options/charts.js#L175). Styling is in [options/options.css](https://github.com/schmich/marinara/blob/master/src/options/options.css). My guess is there's some difference in the styling/CSS classes for the heatmap labels vs. the weekly/daily distrbution chart labels.

**Comment by jamielockie** [comment]
Hey,

Thanks for these steps. I'll take a look tonight and try to get back to you tomorrow. I'll let you know if I need some clarification.

**Comment by schmich** [comment]
I forgot to mention it, but the steps above assume you're running on macOS. Building requires `make`, `m4`, and `jq`. `make` and `m4` should already be installed. For `jq` you can `brew install jq` if you use Homebrew, otherwise, you can [download it directly](https://stedolan.github.io/jq/).

**Comment by jamielockie** [comment]
Okay, awesome.  Thanks for clarifying, I'm on windows and was running into some problems building the manifest.json.  I'll keep working on it today!

Thanks again for letting me take a crack at it!

**Comment by jamielockie** [comment]
Okay, cool. Thanks for the clarification. I'm on Windows. I'm having some trouble in the build process, getting an error ```not found: make-manifest/common.json```. I'll keep working on it today!

**Comment by schmich** [comment]
I will need to get the development process smooth on Windows. I'm not sure why that error is happening. Are you switched over to the Firefox branch (`git checkout firefox`)?

For now, if you want, you can just paste this manifest into `src/manifest.json` (this is the result of the `make dev-firefox` step):

```json
{
  "name": "__MSG_app_name__",
  "short_name": "__MSG_app_name_short__",
  "description": "__MSG_app_desc__",
  "default_locale": "en",
  "version": "2.1.1",
  "manifest_version": 2,
  "author": "Chris Schmich <schmch@gmail.com>",
  "homepage_url": "https://github.com/schmich/marinara",
  "offline_enabled": true,
  "permissions": [
    "contextMenus",
    "storage",
    "notifications"
  ],
  "browser_action": {
    "default_title": "Marinara",
    "default_icon": "icons/browser-action.png"
  },
  "background": {
    "persistent": true,
    "scripts": [
      "lib/EventEmitter.min.js",
      "common/T.js",
      "background/enum.js",
      "background/notifications.js",
      "background/menu.js",
      "background/timer.js",
      "background/sounds.js",
      "background/async.js",
      "background/settings.js",
      "background/rle.js",
      "background/history.js",
      "background/messages.js",
      "background/background.js"
    ]
  },
  "options_page": "options/options.html",
  "icons": {
    "16": "icons/16.png",
    "48": "icons/48.png",
    "128": "icons/128.png"
  },
  "minimum_chrome_version": "55",
  "content_security_policy": "script-src 'self'; object-src 'self'",
  "applications": {
    "gecko": {
      "id": "firefox-addon@marinara.com",
      "strict_min_version": "57.0"
    }
  }
}
```

**Comment by jamielockie** [comment]
Thanks for that.  I was able to get up and running, and fix the font size. As far as I can tell, your ```#heatmap``` was inheriting the base font-size of ```16px``` from the body.  Is the next step for me to take that hard-coded manifest.json out and submit a pull request? This has been fun!

**Comment by schmich** [comment]
👍  Awesome! A pull request would be great. `src/manifest.json` is in [`.gitignore`](https://github.com/schmich/marinara/blob/firefox/.gitignore) because it's dynamically generated and should never be added, so no git tool should ask you to add it.

If your commit message starts with "Fix #114", it will automatically reference this issue, which helps to track which commits fixed which issues.

Thanks again for the help.

**Comment by schmich** [comment]
Fixed with 3b99ff2e481685ff3ea82ceeee0e8b436a8ab57a.
